drivers/flash/soc_flash_nrf_mram.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/barrier.h>
+
+LOG_MODULE_REGISTER(flash_nrf_mram, CONFIG_FLASH_LOG_LEVEL);
+
+#define DT_DRV_COMPAT nordic_mram
+
+#define MRAM_START DT_INST_REG_ADDR(0)
+#define MRAM_SIZE  DT_INST_REG_SIZE(0)
+
+#define MRAM_WORD_SIZE DT_INST_PROP(0, write_block_size)
+#define MRAM_WORD_MASK (MRAM_WORD_SIZE - 1)
+
+#define ERASE_VALUE 0xff
+
+#if defined(CONFIG_SOC_FLASH_NRF_MRAM_ONE_BYTE_WRITE_ACCESS)
+#define WRITE_BLOCK_SIZE 1
+#else
+#define WRITE_BLOCK_SIZE MRAM_WORD_SIZE
```

Comment:
In other words, do you want me to `#define MIN_WRITE_BLOCK_SIZE 1` and check that the DTS value is divisible by it?

Reply:
No. What I mean is that if you do not set `CONFIG_SOC_FLASH_NRF_MRAM_ONE_BYTE_WRITE_ACCESS`, then you already have assumed that we are only allowed to write by some device specific write block size, as a minimal unit.
That is what device accepts, yet we allow to modify this by selecting something in DTS, so if device has write-block-size requirement then any value set by DTS should be multiply of device requirement.

In case if you set CONFIG_SOC_FLASH_NRF_MRAM_ONE_BYTE_WRITE_ACCESS, then you probably do not care what user sets there, any value is multiply of CONFIG_SOC_FLASH_NRF_MRAM_ONE_BYTE_WRITE_ACCESS.

The DTS can independently be used to modify behaviour of code, but should not violate what device can do; for example nrf52840 has 4 byte write-block-size, but user may want to have there 16, this is ok, but if user sets DTS to 2, that is something the device can not do.

Reply:
Having looked at other recent flash driver PRs, I decided to only use the MRAM word size hardcoded in the driver.

I don't think this is ideal, I think constants like this should be fetched from DTS, but there doesn't seem to be an established pattern for that in flash at the moment.

Reply:
Not really. New drivers actually start to use the DTS.
And yest soc_nrf_flash does that, which is pain in the lower parts if you try to adjust the write-block-size above what is hardcoded.
I will PR this change further or later, will not insist on this change here. Marking as resolved.